trunner/config.py

```diff
@@ -157,16 +157,25 @@ def _parse_shell_command(self, config: dict):
             raise ParserError("execute/run attribute cannot be empty")
 
```

Comment:
I think it may be simplified a little bit, not fully tested, but there is my proposal:

```suggestion
        if execute_binary:
            # binary_path may be both an absolute path to a binary or just binary name (then the path will be default)
            binary_path = Path(cmd[0])

            binary = binary_path.name
            path = binary_path.parent if binary_path.is_absolute() else self.ctx.target.exec_dir()
            prefix = str(path) + "/" if self.ctx.target.rootfs else "sysexec "
            parsed_cmd = shlex.split(prefix + binary) + cmd[1:]
        else:
            binary = None
            parsed_cmd = cmd
            path = Path()

        self.test.shell = ShellOptions(
            binary=binary,
            cmd=parsed_cmd,
            path=path.relative_to("/"),
        )
```